crowbar_framework/app/models/service_object.rb

```diff
@@ -889,7 +889,7 @@ def violates_platform_constraint?(elements, role)
         node = NodeObject.find_node_by_name(element)
 
         return true if !constraints.any? do |platform, version|
-          node[:platform] == platform && equal_or_match_version(node[:platform_version], version)
+          node[:platform] == platform && PlatformRequirement.new(version).satisfied_by?(node[:platform_version])
```

Comment:
Could we also add a helper on the `node` object to tidy this up?

    node[:platform] == platform && node.version_matches(version)